src/service.rs

```diff
@@ -1242,7 +1178,22 @@ where
                 ty,
                 listen_address,
             } => {
-                self.session_open(cx, handle, Some(public_key), address, ty, listen_address);
+                if ty.is_outbound() {
+                    self.state.decrease();
+                }
+                let is_continue = self
```

Comment:
suggest to use an encapsulation fn

```suggestion
                if !self.reached_max_connection_limit() {
                    self.session_open(cx, handle, public_key, address, ty, listen_address);
                }
```

and I don't think the size of listen address should be counted into connection limit, for example, user setup a server with 2 listen address and 1 max_connection_number config, in current code, no client are allowed to connect to this server.